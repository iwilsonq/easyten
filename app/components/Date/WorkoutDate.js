import React from 'react';
import { View } from 'react-native';
import { formatWorkoutDate } from '../../libs/helpers';
import { BodyText } from '../Text';

const WorkoutDate = ({ created }) => (
	<View>
		<BodyText>{formatWorkoutDate(created)}</BodyText>
	</View>
);

export { WorkoutDate };
