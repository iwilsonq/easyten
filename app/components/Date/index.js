export { DateHeader } from './DateHeader';
export { WorkoutDate } from './WorkoutDate';
