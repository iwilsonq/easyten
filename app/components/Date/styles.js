import React from 'react';
import g from 'glamorous-native';
import { theme, typography } from '../../config/theme';

const Header = g.text(
	{
		...typography.header
	},
	({ styles, ...props }) => ({
		textAlign: props.centered ? 'center' : 'left',
		...styles
	})
);

const Text = g.text(
	{
		...typography.body
	},
	({ styles, ...props }) => ({
		textAlign: props.centered ? 'center' : 'left',
		...styles
	})
);

const DividerWrapper = g.view({
	width: '100%',
	flex: 1,
	alignItems: 'center'
});

const DividerDash = g.view({
	width: 120,
	height: 1,
	backgroundColor: theme.grey,
	marginVertical: 20
});

const Divider = () => (
	<DividerWrapper>
		<DividerDash />
	</DividerWrapper>
);

export { Header, Text, Divider };
