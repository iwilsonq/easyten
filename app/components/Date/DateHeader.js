import React from 'react';
import { View } from 'react-native';
import { Header, Text, Divider } from './styles';

const DateHeader = props => (
	<View>
		<Header centered>Today</Header>
		<Text centered>Tuesday, January 15, 2018</Text>
		<Divider />
	</View>
);

export { DateHeader };
