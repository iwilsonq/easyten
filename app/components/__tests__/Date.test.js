import React from 'react';
import renderer from 'react-test-renderer';
import { DateHeader } from '../Date';

describe('DateHeader', () => {
	const componentJSON = renderer.create(<DateHeader />).toJSON();

	it('renders properly', () => {
		expect(componentJSON).toMatchSnapshot();
	});
});
