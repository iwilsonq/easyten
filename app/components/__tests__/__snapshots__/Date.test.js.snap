// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateHeader renders properly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    centered={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#dededf",
          "fontFamily": "montserrat-semibold",
          "fontSize": 18,
          "fontWeight": "600",
          "paddingBottom": 8,
        },
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Today
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    centered={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#dededf",
          "fontFamily": "montserrat-regular",
          "fontSize": 14,
        },
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Tuesday, January 15, 2018
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#9e9ea0",
            "height": 1,
            "marginVertical": 20,
            "width": 120,
          },
        ]
      }
    />
  </View>
</View>
`;
